src/Microsoft.AspNetCore.TestHost/TestServer.cs

```diff
@@ -27,6 +27,19 @@ public TestServer(IWebHostBuilder builder)
             _hostInstance = host;
         }
 
+        public TestServer(IWebHostBuilder builder, IFeatureCollection featureCollection)
```

Comment:
I originally tried this:
`server.Features = featureCollection;`
but it says:
> Property or indexer `IServer.Features` cannot be assigned to - it is read only.

That's also why added the property initializer of:
`IFeatureCollection IServer.Features { get; } = new FeatureCollection();`
because I wasn't sure how else to give it a collection to work with.